routers/web/org/times.go

```diff
@@ -0,0 +1,208 @@
+// Copyright 2014 The Gogs Authors. All rights reserved.
+// Copyright 2022 The Gitea Authors.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package org
+
+import (
+	"net/http"
+	"time"
+
+	"code.gitea.io/gitea/models/db"
+	"code.gitea.io/gitea/modules/base"
+	"code.gitea.io/gitea/modules/context"
+	"code.gitea.io/gitea/modules/setting"
+
+	"xorm.io/builder"
+)
+
+const (
+	tplByRepos      base.TplName = "org/times/times_by_repos"
+	tplByMembers    base.TplName = "org/times/times_by_members"
+	tplByMilestones base.TplName = "org/times/times_by_milestones"
+)
+
+// parseTimes contains functionality that is required in all these functions,
+// like parsing the date from the request, setting default dates, etc.
+func parseTimes(ctx *context.Context) (unixfrom, unixto int64, err error) {
+	// View variables
+	ctx.Data["PageIsOrgTimes"] = true
+	ctx.Data["AppSubURL"] = setting.AppSubURL
+
+	// Time range from request, if any
+	from := ctx.FormString("from")
+	to := ctx.FormString("to")
+	// Defaults for "from" and "to" dates, if not in request
+	if from == "" {
+		// DEFAULT of "from": start of current month
+		from = time.Now().Format("2006-01") + "-01"
+	}
+	if to == "" {
+		// DEFAULT of "to": today
+		to = time.Now().Format("2006-01-02")
+	}
+
+	// Prepare Form values
+	ctx.Data["RangeFrom"] = from
+	ctx.Data["RangeTo"] = to
+
+	// Prepare unix time values for SQL
+	gfrom, err := time.Parse("2006-01-02", from)
+	if err != nil {
+		ctx.ServerError("TimesError", err)
+	}
+	unixfrom = gfrom.Unix()
+	gto, err := time.Parse("2006-01-02", to)
+	if err != nil {
+		ctx.ServerError("TimesError", err)
+	}
+	// Humans expect that we include the ending day too
+	unixto = gto.Add(1440 * time.Minute).Unix()
+	return
+}
+
+// TimesByRepos renders worktime by repositories.
+func TimesByRepos(ctx *context.Context) {
+	// Run common functionality
+	unixfrom, unixto, err := parseTimes(ctx)
+	if err != nil {
+		return
+	}
+
+	// Set submenu tab
+	ctx.Data["TabIsByRepos"] = true
+
+	// Get the data from the DB
+	type result struct {
+		Name    string `xorm:"name"`
+		SumTime int64  `xorm:"sumtime"`
+	}
+	sql, args, err := builder.Select("repository.name", "SUM(tracked_time.time) AS sumtime").
+		From("tracked_time").
+		InnerJoin("issue", "tracked_time.issue_id = issue.id").
+		InnerJoin("repository", "issue.repo_id = repository.id").
+		Where(builder.Eq{"repository.owner_id": ctx.Org.Organization.ID}).
+		And(builder.Eq{"tracked_time.deleted": 0}).
```

Comment:
```suggestion
		And(builder.Eq{"tracked_time.deleted": false}).
```

And same in other places

Reply:
Thanks, will do!